packages/dds/tree/src/simple-tree/api/schemaFactoryRecursive.ts

```diff
@@ -117,21 +117,28 @@ export function createFieldSchemaUnsafe<
  * Be very careful when declaring recursive schema.
  * Due to the removed extends clauses, subtle mistakes will compile just fine but cause strange errors when the schema is used.
  *
- * For example if the square brackets around the allowed types are forgotten:
+ * For example if a reference to a schema is malformed (in this case boxed inside an object):
  *
  * ```typescript
- * class Test extends sf.arrayRecursive("Test", () => Test) {} // Bad
+ * class Test extends sf.arrayRecursive("Test", [() => ({ Test })]) {} // Bad
```

Comment:
The old example here actually gives an error even without ValidateRecursiveSchema due to improved recursive typing from recent changes. This adjustment returns it to a case which behaves as documented here.